platforms/software/ivy/src/main/java/org/gradle/api/publish/ivy/tasks/GenerateIvyDescriptor.java

```diff
@@ -124,4 +90,29 @@ private static IvyModuleDescriptorSpecInternal toIvyModuleDescriptorInternal(Ivy
         }
     }
 
+    @Inject
+    protected abstract ObjectFactory getObjectFactory();
+
+    @Inject
+    @Deprecated
+    @SuppressWarnings("DeprecatedIsStillUsed")
+    protected abstract PathToFileResolver getFileResolver();
+
+    static class GenerateIvyDescriptorAdapter {
+        @BytecodeUpgrade
+        static File getDestination(GenerateIvyDescriptor self) {
+            return self.getDestination().getAsFile().getOrNull();
+        }
+
+        @BytecodeUpgrade
+        static void setDestination(GenerateIvyDescriptor self, File destination) {
+            self.getDestination().fileValue(destination);
+        }
+
+        @BytecodeUpgrade
+        static void setDestination(GenerateIvyDescriptor self, Object destination) {
+            ProviderApiDeprecationLogger.logDeprecation(GenerateIvyDescriptor.class, "setDestination(Object)", "getDestination()");
```

Comment:
Should we add a test that verifies that this deprecation is emitted? Is that even possible with current bytecode upgrade infrastructure?